### PR TITLE
feat(ci): the beginnings of some release automation

### PR DESCRIPTION
**Pull request by sanderblue**
This PR introduces a new `make release` target to facilitate the release process. The [runbook](https://source.datanerd.us/developer-toolkit/runbooks/wiki/New-Relic-Go-SDK) has also been updated.

Example `make` command:
- `make release version=0.11.0`

### CODE REVIEW

File: build/release.mk [hunk 1]
```diff
@@ -0,0 +1,6 @@
+RELEASE_SCRIPT_PATH ?= ./tools/release.sh
+
+# Example usage: make release version=0.11.0
```
Comment (zlesnr) [inline build/release.mk]: Is there a way that we can require `version` as an argument?  This is me not knowing make well enough.

File: build/release.mk [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+release:
+	@echo "=== $(PROJECT_NAME) === [ release          ]: Generating release."
+	./tools/release.sh $(version)
```
Comment (zlesnr) [inline build/release.mk]: Maybe use `RELEASE_SCRIPT_PATH` from above in place of the direct path.  What do you think?

File: build/release.mk (hunk 1 above)
Comment (sanderblue) [reply build/release.mk]: It is required. If it's not passed in, an error message is displayed to the user.

File: build/release.mk (hunk 2 above)
Comment (sanderblue) [reply build/release.mk]: Oh yeah, I meant to do that (jet lag fever and such 😉). Thanks.

File: build/release.mk (hunk 1 above)
Comment (zlesnr) [reply build/release.mk]: Right, of course.

File: build/release.mk (hunk 2 above)
Comment (sanderblue) [reply build/release.mk]: Updated

**Comment by zlesnr** [comment]
Nice work @sanderblue.  Thanks for putting this up.

**Pull request merged**